cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java

```diff
@@ -137,15 +140,24 @@ protected void setupLaunchConfig(ProgramLaunchConfig launchConfig, Program progr
     ApplicationSpecification appSpec = program.getApplicationSpecification();
     SparkSpecification spec = appSpec.getSpark().get(program.getName());
 
-    Map<String, String> clientArgs = RuntimeArguments.extractScope("task", "client",
-                                                                   options.getUserArguments().asMap());
+    Arguments userArguments = options.getUserArguments();
+    Map<String, String> clientArgs = RuntimeArguments.extractScope("task", "client", userArguments.asMap());
     // Add runnable. Only one instance for the spark client
     launchConfig.addRunnable(spec.getName(), new SparkTwillRunnable(spec.getName()), 1,
                              clientArgs, spec.getClientResources(), 0);
 
     Map<String, String> extraEnv = new HashMap<>();
     extraEnv.put(Constants.SPARK_COMPAT_ENV, sparkCompat.getCompat());
 
+    if (!userArguments.hasOption(SystemArguments.RUNTIME_MANAGER_TYPE) ||
```

Comment:
What is this for?

Reply:
Had offline discussion about this condition. Made changes as per the discussion. Testing manually now.

Reply:
Had offline discussion about this condition. Made changes as per the discussion. Testing manually now.